SDK_versioned_docs/version-3.0.0/widgets/swap-widget-api.md

```diff
@@ -30,6 +30,7 @@ sidebar_position: 3
 | `locale`                    | `SupportedLocale`             | `en-US`                 | Specifies an explicit locale to use for the widget interface. This can be set to one of the values exported by the library in [`SUPPORTED_LOCALES`](https://github.com/Uniswap/widgets/blob/main/src/constants/locales.ts).                                                                                                                                                                                                                                                                                                        |
 | `onConnectWallet`           | `() => void`                  | `undefined`             | If passed, the “Connect your wallet” message will be clickable, and clicking it will trigger this handler function. This can be used to trigger your own wallet connection flow from the widget.                                                                                                                                                                                                                                                                                                                                     |
 | `onError`                   | `ErrorHandler`                | `undefined`             | An error handler which receives any errors that occur in the widget. This can be used for collecting error metrics.                                                                                                                                                                                                                                                                                                                                                                                                                  |
+| `onTokenSelectorClick` | `() => void \| () => Promise<boolean \| void>` | `undefined` | A click handler fired when the user clicks on the token selector dropdown. This may be useful if you would like to utilize your own external token selector UI.                                                                                                                                                                                                                                                                                                                                                                                                                  |
```

Comment:
should this specify whether the input or output token selector was clicked?

Reply:
i.e. `(field: Field) => void \| (field: Field) => Promise<boolean \| void>`

Reply:
Can we give a little more details on what a function that returns void vs. a promise does?

Reply:
maybe  "A click handler fired when the user clicks on the token selector dropdown. To manage progression to the native token selector view (i.e. to utilize your own external token selector UI), return a resolved promise with resolve(true/false)." ?